src/datasources/flow-ds-react/helpers.ts

```diff
@@ -945,3 +951,38 @@ export const getFilteredNodes = async ({ client, simpleRequest }, exporterNodes?
     }
 }
 
+export const convertTemplateVariables = (item: FlowQueryData, templateSrv) => {
+    item.functions?.forEach((f, idx) => {
+
+        if (item.functionParameters[idx]) {
+            item.functionParameters[idx] = templateSrv.replace(item.functionParameters[idx])
+
+        } if (item.parameterOptions[idx] && item.parameterOptions[idx].value) {
+            item.parameterOptions[idx].value = templateSrv.replace(item.parameterOptions[idx].value)
+            item.parameterOptions[idx].label = templateSrv.replace(item.parameterOptions[idx].label)
+        }
+
+    })
+    return item
+}
+
+export const lookupIfIndex = async (nodeQuery: any, iface: any, simpleRequest) => {
```

Comment:
Some additional documentation is required here. I get the feeling this code is directly copied from the older version of Helm. We at least need to get rid of the uses of `any` to bypass Typescript, and if we could break up the deep levels of indentation that would be a positive as well.

Converting the function to a single return statement (with a `break;` to interrupt the for loop) at the end might also help future debugging.

Reply:
ah this, came from [this](https://opennms.atlassian.net/browse/HELM-333) jira which I believe using template variables from different datasources would't  work.I think this should be handled differently eventually (like a common standard call ) but in the mean time I'd keep it like it is because it's more complicated than it looks (I think).